### ISSUE TITLE
mid-bootcamp-project_Yuri Awano

### ISSUE THREAD

**Issue by yuawano** [Ironhack-Data-0621-Remote/mid-bootcamp-project#17]
https://github.com/yuawano/mid-bootcamp-project

**Comment by Fominayasg** [comment]
# Hi Yuri 😁 !!

In general your project is very well developed although I would organize it a little different. Anyways I'll deep a little bit in details to give you some (I hope useful) advice.

## README file
Your Readme is very well explained but maybe a too long, this is just the "teaser" to your project, of course you can keep it like this, is a matter of taste, but I'd sum up it a little bit.

You could leave just the essential information, and keep out of it the developmente of the procedures (Exploring the data in SQL, Data exploration and modeling in python). You can mention it inviting the reader to jump into you jupyter notebooks if they're interested in it, and put all the information you wrote here in your jupyters in a markdown cell. 

Remember that the readme is something that a recruiter will read sometiems wothout any knowledge about python or sql.

In the part where you display the libraries used for the project you don't need to specify all the parts of sklearn that you used. It could be something like this.

```
- [Pandas](https://pandas.pydata.org/docs/)
- [Matplotlib](https://matplotlib.org/)
- [Sklearn](https://scikit-learn.org/stable/#)
- [Seaborn](https://seaborn.pydata.org/)
```
That will look like this:
- [Pandas](https://pandas.pydata.org/docs/)
- [Matplotlib](https://matplotlib.org/)
- [Sklearn](https://scikit-learn.org/stable/#)
- [Seaborn](https://seaborn.pydata.org/)

## Repo structure
When you make a project, even if it is for a course like this one, is better not to leave there the "question" files, you can erase that from you project if you want.

Moreover it is important that you use a .gitignore file so you don't upload all those .DS_Store and other temporal files that don't serve any purpose and take up memory in github. For that is important that you create the `.gitignore` file before commiting any other changes.
I see you have one but don't have DS_Store included.

About the folders we use de "data" folder only to store the data itself (csv files for example) 

You can put the jupyter noteboook in the main folder and maybe the models in another one. 

This little details make the reader's job much more easier, which is important if someone has to read a lot of them for example for a hiring process. More over, it talks about your organization skills which is something that always will get you a few positive points.

And finally I would split your jupyter notebook in a couple of them, maybe one for cleaning and exploring and another one for preprocessing and modelling. But this is totally up to you.

## SQL
### Question 5
I think you forgot to limit your results to 10, because I'm pretty sure you know how to do it, but just in case it would be like this 
```SQL
alter table credit_card_data
drop column Q4_Balance
select * from credit_card_data
limit 10;
```

### Question 9

It is totally fine but I leave you here another way to do it:

```SQL
select round(avg(Average_Balance),2) from credit_card_data;
```

## EDA & Data cleaning
You made a reaally good job here. I specially liked how you treated the null values.

## Data preprocessing
You analized exactly what is needed, it's pretty clear that you understand the purpose of each procedure (encoding, rescaling...)

## ML

Just a little disclaimer here,  make sure that you don't get confused with linear and logistic regression (I know the names are too similar).
In the conclusions part you talk about linear regression while you used logistic regression, be careful with this beacuse in a classification model a linear regression would make no sense and it's a pity that after all this work when the reader gets to the conclusions finds this typo.

In addition if you feel curious you can check some other classification models that exist and see if them works better or worse in this case.
## Tableau
You made some very good visualizations here. Maybe you could add a dashboard where you put it all together explaining it a little more, but I'm really impressed that you created a story 😁 .
## Last tips

To improve your project even more I would recommend to comment the jupyter notebook a little more (specially the visualizations) and (if you have time) try to separate the functions in a .py file so you can importe and use them in any file. 

To sum up, it's a really good project and you got to some interesting conclusions. Good job!

PS: I think you upload two images by error in your repository. Maybe you want to check it.
